### PR TITLE
Update KFP prebuilt components pipeline

### PR DESCRIPTION
**Pull request by takumiohym**
- Removed `v2` namespace from KFP.
- Changed `google-cloud-pipeline-components` path for v2.
- Added [`retrieve_best_hptune_component.py`](https://github.com/GoogleCloudPlatform/asl-ml-immersion/blob/d42de7110032c56983b5485e54423e1a7fb166f1/notebooks/kubeflow_pipelines/pipelines/solutions/pipeline_vertex/retrieve_best_hptune_component.py) and Python lightweight component to replace the three deprecated components.
- Added an importer node for custom model load due to the [`ModelUploadOp`](https://google-cloud-pipeline-components.readthedocs.io/en/google-cloud-pipeline-components-2.8.0/api/v1/model.html?h=modeluploadop#v1.model.ModelUploadOp) change.
- Changed JSON to YAML since now it is the [preferred format in KFP v2](https://www.kubeflow.org/docs/components/pipelines/v2/migration/#pipeline-package-file-extension).

### New pipeline
image

### Old pipeline
image

### CODE REVIEW

File: notebooks/kubeflow_pipelines/pipelines/solutions/kfp_pipeline_vertex_prebuilt.ipynb [hunk 1]
```diff
@@ -195,19 +195,20 @@
     "import os\n",
```
Comment (BenoitDherin) [inline notebooks/kubeflow_pipelines/pipelines/solutions/kfp_pipeline_vertex_prebuilt.ipynb:195]: <pre class="ql-syntax" spellcheck="false">Line #32.    from retrieve_best_hptune_component import retrieve_best_hptune_result
</pre><p>Add a blank line separating own imports from 3rd party imports (pep8): i.e. a blank line above this line</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/439/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

**Review by BenoitDherin — APPROVED** [review]
LGTM

File: notebooks/kubeflow_pipelines/pipelines/solutions/kfp_pipeline_vertex_prebuilt.ipynb (hunk 1 above)
Comment (takumiohym) [reply notebooks/kubeflow_pipelines/pipelines/solutions/kfp_pipeline_vertex_prebuilt.ipynb:195]: I personally agree, but actually it is the one pre-commit prefers (no blank line). It keeps reverting when we add a blank line between them.

File: notebooks/kubeflow_pipelines/pipelines/solutions/pipeline_vertex/pipeline_prebuilt.py [hunk 2]
```diff
@@ -15,19 +15,20 @@
...
-from google_cloud_pipeline_components.v1.custom_job import CustomTrainingJobOp
 from google_cloud_pipeline_components.v1.hyperparameter_tuning_job import (
     HyperparameterTuningJobRunOp,
+    serialize_metrics,
+    serialize_parameters,
 )
-from kfp.v2 import dsl
+from google_cloud_pipeline_components.v1.model import ModelUploadOp
+from kfp import dsl
+from retrieve_best_hptune_component import retrieve_best_hptune_result
```
[the remark this answers is not in the document]
Comment (takumiohym) [reply notebooks/kubeflow_pipelines/pipelines/solutions/pipeline_vertex/pipeline_prebuilt.py:31]: I personally agree, but actually it is the one pre-commit prefers (no blank line). It keeps reverting when we add a blank line between them.

**Pull request merged**